### PR TITLE
API-39630-handle-brd-timeouts

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Adds lighthouse common exception for bad gateway. 
- Adds catch for external timeout and returns a bad gateway in custom error. 
- Adds a corresponding test. Adds the bad_gateway to lighthouse_error_handler, and corresponding test.

## Related issue(s)

- [API-39630](https://jira.devops.va.gov/browse/API-39630)

## Testing done

- [X] *New code is covered by unit tests*

## What areas of the site does it impact?
	new file:   modules/claims_api/lib/claims_api/common/exceptions/lighthouse/bad_gateway.rb
	modified:   modules/claims_api/lib/claims_api/v2/error/lighthouse_error_handler.rb
	modified:   modules/claims_api/lib/custom_error.rb
	modified:   modules/claims_api/spec/lib/claims_api/v2/error/lighthouse_error_handler_spec.rb
	modified:   modules/claims_api/spec/sidekiq/claim_custom_error_spec.rb

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

Ideas for testing

### CODE REVIEW

File: modules/claims_api/spec/sidekiq/claim_custom_error_spec.rb [hunk 1]
```diff
@@ -164,4 +164,35 @@
       end
     end
   end
+
+  context 'an external service returns a 504' do
+    error_original_body = {
+      messages: [
+        {
+          'key' => 'timout',
```
Comment (tycol7) [inline modules/claims_api/spec/sidekiq/claim_custom_error_spec.rb:172]: sp: `timeout`

**Review by tycol7 — APPROVED** [review]
lgtm!

**Review by tycol7 — CHANGES_REQUESTED** [review]
oops, a couple [failing tests](https://github.com/department-of-veterans-affairs/vets-api/actions/runs/10622029432/job/29498556482)

**Comment by stiehlrod** [comment]
> oops, a couple [failing tests](https://github.com/department-of-veterans-affairs/vets-api/actions/runs/10622029432/job/29498556482)

@tycol7  Those failing tests belong to benefits_documents; none of my tests are failing.

**Review by tycol7 — APPROVED** [review]
Must've been a flaky test.

**Pull request merged**